eth/watchers/roundswatcher.go

```diff
@@ -136,6 +149,9 @@ func (rw *RoundsWatcher) handleLog(log types.Log) error {
 		return fmt.Errorf("eventName is not NewRound")
 	}
 
+	fmt.Println(log)
+	rw.subFeed.Send(log)
```

Comment:
I think it makes sense to treat the event the log is sent over to the SenderWatcher as an 'event channel' even though this is not fully the case. 

By this I mean that we do not use the crucial value pertaining to the emitted event, which would be the round number. We only check whether the event was removed or not to branch into different logic, receiving a message on the channel itself would indicate a new round event already. This seems to be the most effective because otherwise we would be sending _every_  event through the channel and would still have to check the signature in `SenderWatcher`

Reply:
To clarify, I'm not saying that we shouldn't send the log into the feed. I was suggesting that we send the log to the feed after checking if the log successfully decodes into a `NewRound` event. We already check that the log's first topic hash maps to the `NewRound` event name so this would just be an extra sanity check to make sure that the log actually decodes into the Go struct and that the included data is not malformed.

Reply:
Ahh yes sorry, I see what you mean :) !